### PR TITLE
nwchem: support Fujitsu A64FX compilers

### PR DESCRIPTION
**Pull request by h-murai**
Some patches are added to compile NWChem with Fujitsu compilers. These are discussed in https://github.com/nwchemgit/nwchem/issues/347 and to be merged to NWChem.

### CODE REVIEW

File: var/spack/repos/builtin/packages/nwchem/package.py [hunk 1]
```diff
@@ -27,6 +27,8 @@ class Nwchem(Package):
     variant('openmp', default=False, description='Enables OpenMP support')
     variant('mpipr', default=False, description='Enables ARMCI with progress rank')
 
+    patch('fj.patch', when='@7.0.2 %fj')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/nwchem/package.py:34]: Can you explain in a brief comment above the directive why the patch is needed and leave a reference to the issue in the description?

**Review by alalazo — CHANGES_REQUESTED** [review]
LGTM. Only a minor request.

File: var/spack/repos/builtin/packages/nwchem/package.py (hunk 1 above)
Comment (h-murai) [reply var/spack/repos/builtin/packages/nwchem/package.py:34]: This patch handles the modification to the build system (e.g. compiler flags) and Fortran language syntax, and enables Fujitsu compilers to compile NWChem. You  can find the detailed discussion about it  in https://github.com/nwchemgit/nwchem/issues/347 .

Comment (alalazo) [reply var/spack/repos/builtin/packages/nwchem/package.py:34]: Thanks @h-murai The request was to add this info in a code comment above the patch (so the next developer knows what to expect in there). Then the PR LGTM.

**Comment by h-murai** [comment]
@alalazo I added the description of the patch.

**Review by alalazo — APPROVED** [review]

**Pull request merged**